source_gen/test/builder_test.dart

```diff
@@ -84,6 +84,25 @@ void main() {
     );
   });
 
+  test('Omits generated comments if writeDescriptions is explicitly false',
+      () async {
+    final srcs = _createPackageStub();
+    final builder = LibraryBuilder(
+      const CommentGenerator(),
+      header: '',
+      writeDescriptions: false,
+    );
+    await testBuilder(
+      builder,
+      srcs,
+      generateFor: {'$_pkgName|lib/test_lib.dart'},
+      outputs: {
+        '$_pkgName|lib/test_lib.g.dart':
+            decodedMatches(isNot(startsWith('// ***'))),
```

Comment:
It would better to check for the exact expected content in this case - if we decided to change the formatting of the description this test wouldn't start failing, but would no longer be doing any good.

Reply:
Makes sense

Reply:
I corrected the test and added another test to sanity check when the default value is unchanged and the generated comments should be present.

I also rewrote the inline docs for `writeDescriptions` to match the style present in the rest of the inline docs.

Reply:
LGTM, thanks